### PR TITLE
Item Router

### PR DESCRIPTION
**Pull request by cynthia-doc**

### CODE REVIEW

File: server/api/itemRouter.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const express = require('express')
+const mongoose = require('mongoose');
+
+const router = express.Router()
+
+router.post('/addItem', (req, res) => {
+    const body = req.body;
+
+    if(!body) {
+        next(err);
```
Comment (Sherry2001) [inline server/api/itemRouter.js:13]: error's not defined here

File: server/api/itemRouter.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    item.save().then(() => {
+        res.status(201).json({
+            success: true,
+            message: 'Item added'
+        })
+    });
+})
+
+router.get('/getItems', async (req, res) => {
+    await User.find({_id:req.body.userId }, (err, item) => {
```
Comment (Sherry2001) [inline server/api/itemRouter.js:31]: don't think await is needed here?

Comment (Sherry2001) [reply server/api/itemRouter.js:31]: oh and findOne() for consistency?

File: server/api/itemRouter.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        return res.json(item);
+    }).catch(err => next(err));
+})
+
+router.put('/addItemToCategory', (req, res) => {
+    const body = req.body;
+    
+    if(!body) {
+        next(err);
```
Comment (Sherry2001) [inline server/api/itemRouter.js:43]: same, don't think err is defined here

Comment (cynthia-doc) [reply server/api/itemRouter.js:43]: ahhhh haha v good catch, I'll fix this and the one earlier

File: server/api/itemRouter.js (hunk 2 above)
Comment (cynthia-doc) [reply server/api/itemRouter.js:31]: ah I seeeee! you right cuz all items are returned at once

File: server/api/itemRouter.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    
+    Item.findOne({_id:body.itemId}, (err, item) => {
+        if (err) {
+            next(err);
+        }
+        item.categoryIds.push(body.categoryId);
+        item.save(done);
+        res.status(200).json({ success: true });
+    })
+
```
Comment (Sherry2001) [inline server/api/itemRouter.js:54]: I think this part miight be a problem?? I think maybe it should be Item.findOne().then(()=> Categroy.findOne() with one big catch() and one res.json for both. I'll try to edit and u can check if it makes sense?

Comment (cynthia-doc) [reply server/api/itemRouter.js]: i just changed it if you havent done so!

Comment (Sherry2001) [reply server/api/itemRouter.js]: oh shoot i didn't see...

File: server/api/itemRouter.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    item.save().then(() => {
+        res.status(201).json({
+            success: true,
+            message: 'Item added'
+        })
+    });
+})
+
+router.get('/getItems', async (req, res) => {
+    await User.find({_id:req.body.userId }, (err, item) => {
```
Comment (Sherry2001) [inline server/api/itemRouter.js:31]: Item.find

File: server/api/itemRouter.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
+const Item = require('../db/models/item')
```
Comment (Anthony-Fiddes) [inline server/api/itemRouter.js:1]: Add autocomplete for semicolons to your linter

**Pull request merged**